coreblocks/priv/csr/csr_instances.py

```diff
@@ -78,6 +82,18 @@ def __init__(self, gen_params: GenParams):
         mepc_ro_bits = 0b1 if Extension.C in gen_params.isa.extensions else 0b11  # pc alignment (SPEC)
         self.mepc = CSRRegister(CSRAddress.MEPC, gen_params, ro_bits=mepc_ro_bits)
 
+        self.priv_mode = CSRRegister(
+            None,
+            gen_params,
+            width=2,
```

Comment:
Not something for this PR, but what about accepting arbitrary shapes in CSR definitions? This would allow, e.g., using strongly typed `Enum` instead of weakly typed `IntEnum` more.
For now, I would write:
```suggestion
            width=Shape.cast(PrivilegeLevel),
```

Reply:
I'm not sure if that would be too beneficial, as CSR can be written with arbitrary values from functional unit and typing could be wrong in some cases.